.github/workflows/check.yml

```diff
@@ -0,0 +1,31 @@
+name: Check
+
+on:
+    pull_request:
+        branches: [ main, stable-* ]
+    push:
+        branches: [ main, stable-* ]
+
+jobs:
+    check:
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                task: [ detekt, spotlessKotlinCheck, lint ]
+        steps:
+            -   uses: actions/checkout@v3
+            -   name: Set up JDK 11
+                uses: actions/setup-java@v3
+                with:
+                    distribution: "temurin"
+                    java-version: 11
+            -   name: Check ${{ matrix.task }}
+                uses: gradle/gradle-build-action@v2
+                with:
+                    arguments: ${{ matrix.task }}
+            -   uses: actions/upload-artifact@v3
+                if: ${{ always() }}
+                with:
+                    name: ${{ matrix.task }}-report
+                    path: library/build/reports
```

Comment:
Any reason to remove it? It gets uploaded to GH and auto-deleted after 90 days. It's faster to check than running things locally.

I don't want to bother uploading things to nc.k.m right now, this is a quick setup.

re: nc.k.m -> check https://github.com/nextcloud/android/pull/10345 for an alternative.

Reply:
That being said, I just pushed a commit to fix the upload path, `library/build/reports` is not correct

Reply:
Okay :)
Fine by me, in long run we need to find another way than nc.k.m, so if we already get used to this approach, it might be one extra step, but still the right thing to do.